test/error_test.exs

```diff
@@ -0,0 +1,143 @@
+    test "has a context field populated when there is a list of errors" do
+      test_error1 = TestError.exception(some_field: :a)
+      test_error2 = TestError.exception(some_field: :b)
+
+      err = Ash.Error.to_error_class([test_error1, test_error2], error_context: "some context")
+
+      assert err.error_context == ["some context"]
+    end
+
+    test "accumulates error_context field in child errors" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil, error_context: "some other context")
+
+      error_class =
+        Ash.Error.to_error_class([error1, error2], error_context: "some higher context")
+
+      child_error_1 = Enum.find(error_class.errors, fn err -> err.error == "whoops!" end)
+      assert child_error_1.error_context == ["some higher context", "some context"]
+
+      child_error_2 = Enum.find(error_class.errors, fn err -> err.error == "whoops, again!!" end)
+      assert child_error_2.error_context == ["some higher context", "some other context"]
+    end
+
+    test "accumulates error_context field in child errors who have no error_context of their own" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil)
+
+      error_class =
+        Ash.Error.to_error_class([error1, error2], error_context: "some higher context")
+
+      child_error_1 = Enum.find(error_class.errors, fn err -> err.error == "whoops!" end)
+      assert child_error_1.error_context == ["some higher context", "some context"]
+
+      child_error_2 = Enum.find(error_class.errors, fn err -> err.error == "whoops, again!!" end)
+      assert child_error_2.error_context == ["some higher context"]
+    end
+
+    test "leaves child error contexts unchanged if no error_context field provided" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil, error_context: "some other context")
+
+      error_class = Ash.Error.to_error_class([error1, error2])
+
+      child_error_1 = Enum.find(error_class.errors, fn err -> err.error == "whoops!" end)
+      assert child_error_1.error_context == ["some context"]
+
+      child_error_2 = Enum.find(error_class.errors, fn err -> err.error == "whoops, again!!" end)
+      assert child_error_2.error_context == ["some other context"]
+    end
+
+    test "error message contains error context breadcrumbs" do
+      error1 = Ash.Error.to_ash_error("whoops!", nil, error_context: "some context")
+      error2 = Ash.Error.to_ash_error("whoops, again!!", nil, error_context: "some other context")
+
+      error_class =
+        Ash.Error.to_error_class([error1, error2], error_context: "some higher context")
+
+      error_message = Ash.Error.Unknown.message(error_class)
+
+      assert error_message =~ "Context: some higher context > some context"
```

Comment:
Not sure if these assertions would be considered a bit brittle.
My intent here is that we can write some tests that specify / exemplify the format we want the context to take when rendered in the error message.

Reply:
I think for now this is a great starting point. If they prove to be brittle for whatever reason we can change them